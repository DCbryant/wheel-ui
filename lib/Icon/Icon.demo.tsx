import React, {Fragment} from 'react';
import Demo from '../../demo';
import IconExample2 from './Icon.example';

const IconDemo = () => {
  return (
    <Fragment>
      <Demo code={require('!!raw-loader!./Icon.example.tsx').default}>
        <IconExample2/>
      </Demo>
    </Fragment>
  );
};

export default IconDemo;