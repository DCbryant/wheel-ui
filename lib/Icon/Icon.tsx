import React from 'react'
import '../icons/wechat.svg'
import '../icons/qq.svg'
import '../icons/aliPay.svg'
// import './importIcons'

import classes from '../helpers/classes';
import './icon.scss';

interface IconProps extends React.SVGAttributes<SVGElement> {
  name: string;
}

const Icon: React.FunctionComponent<IconProps> = ({className, name, ...restProps}) => {
  console.log(classes('wheel-icon', className))
  return (
    <span>
      <svg className={classes('wheel-icon', className)} {...restProps}>
        <use xlinkHref={`#${name}`}></use>
      </svg>
    </span>
  )
}

export default Icon