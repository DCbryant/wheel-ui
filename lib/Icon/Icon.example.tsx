import React from 'react'
import Icon from './Icon'

const IconExample : React.FunctionComponent = () => {
  return (
    <div>
      <Icon name='wechat'></Icon>
      <Icon name='aliPay'></Icon>
      <Icon name='qq'></Icon>
    </div>
  )
}

export default IconExample