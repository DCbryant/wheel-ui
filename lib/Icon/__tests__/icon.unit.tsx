import * as renderer from 'react-test-renderer'
import {mount} from 'enzyme'
import React from 'react' 
import Icon from '../Icon'

describe('icon', () => {
  it('icon render successful', () => {
    const json = renderer.create(<Icon name='wechat' />).toJSON()
    expect(json).toMatchSnapshot()
  })

  it('icon onClick', () => {
    const fn = jest.fn()
    // const fn2 = jest.fn()
    
    const component = mount(<Icon name='wechat' onClick={fn} />)
    component.find('svg').simulate('click')
    expect(fn).toBeCalled()
  })
})