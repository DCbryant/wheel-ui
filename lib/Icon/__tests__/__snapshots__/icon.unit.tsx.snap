// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon icon render successful 1`] = `
<span>
  <svg
    className="wheel-icon"
  >
    <use
      xlinkHref="#wechat"
    />
  </svg>
</span>
`;
